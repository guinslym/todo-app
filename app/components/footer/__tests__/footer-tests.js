import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import Footer from '../index';

const mockStore = configureStore();

describe('<Footer/>', () => {
  let component;

  beforeEach(() => component = shallow(<Footer />));

  it('should render', () => {
    expect(component.length).toBeTruthy();
  });

  it('should display the footer text', () => {
    expect(component.find('p')).toBePresent();
  });

});

describe('<Footer/> snapshot', () => {

  it('should render correctly', () => {
    let store = mockStore([]);
    const tree = renderer.create(<Provider store={store}><Footer /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
