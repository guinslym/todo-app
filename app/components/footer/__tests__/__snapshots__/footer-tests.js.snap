// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> snapshot should render correctly 1`] = `
<p>
  Show:
   
  <a
    href="#"
    onClick={[Function]}
  >
    All
  </a>
   | 
  <a
    href="#"
    onClick={[Function]}
  >
    Active
  </a>
   | 
  <a
    href="#"
    onClick={[Function]}
  >
    Completed
  </a>
</p>
`;
